plugins/modules/cloud_resource_type.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2024, Network to Code (@networktocode) <info@networktocode.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: cloud_resource_type
+short_description: Creates or removes cloud_resource_type from Nautobot
+description:
+  - Creates or removes cloud_resource_type from Nautobot
+notes:
+  - Tags should be defined as a YAML list
+  - This should be ran with connection C(local) and hosts C(localhost)
+author:
+  - Travis Smith (@tsm1th)
+requirements:
+  - pynautobot
+version_added: "5.4.0"
+extends_documentation_fragment:
+  - networktocode.nautobot.fragments.base
+  - networktocode.nautobot.fragments.tags
+  - networktocode.nautobot.fragments.custom_fields
+options:
+  name:
+    description:
+      - The name of the cloud_resource_type
+    required: true
+    type: str
+  description:
+    description:
+      - The description of the cloud_resource_type
+    required: false
+    type: str
+  cloud_provider:
+    aliases:
+      - provider
+    description:
+      - Required if I(state=present) and the cloud_resource_type does not exist yet
+    required: false
+    type: raw
+  content_types:
+    description:
+      - Cloud Resource Type content type(s). These match app.endpoint and the endpoint is singular.
+      - e.g. cloud.cloudnetwork, cloud.cloudservice (more can be found in the examples)
```

Comment:
> (more can be found in the examples)

I only see 2 options when viewing the Nautobot UI and you have them both listed here. The examples only show 1 option, so not sure if you meant to reword this or something.

Reply:
We should probably also include this as usual:
```
Required if I(state=present) and the cloud resource type does not exist yet
```